lib/src/metta/runner/arithmetics.rs

```diff
@@ -107,7 +107,9 @@ impl Grounded for Bool {
     }
 
     fn match_(&self, other: &Atom) -> MatchResultIter {
-        match_by_equality(self, other)
+        //TODO: THis is a stop-gap until we have a more robust value-bridging system in place
+        // https://github.com/trueagi-io/hyperon-experimental/issues/351
+        match_by_string_equality(&self.to_string(), other)
     }
```

Comment:
I ended up addressing this in https://github.com/trueagi-io/hyperon-experimental/pull/580/commits/5b46301c8b32afa99c8f182a61fb8716aa9c700b  The fix has the advantage that it also addresses a rust grounded atom trying to match a python atom, instead of just the reverse.

All that being said, I'm still a little uneasy about the performance of wrapping simple python objects like a bool with a grounded atom.

Alexey has a good point about not automatically down-converting when we consider big python types or types with differing behaviors (`int` being the primary example), but the cost of crossing the python bridge to check a bool doesn't sit well with me.

I would like to address this in the future.